### ISSUE TITLE
Server starts crashing clients and refusing new connections on startup

### ISSUE THREAD

**Issue by ProfanedBane** [d3athrow/vgstation13#19336]
#### Description of issue
Dreamseeker starts crashing and no one can connect to the server occasionally when the server starts.

#### Difference between expected and actual behavior
EXPECTED: Time for spessmens
ACTUAL: Dead server

#### Steps to reproduce 
Unknown

#### Length of time in which bug has been known to occur

Apparently it's only been the past few weeks.

#### Issue bingo

Please check whatever applies. More checkboxes checked increase your chances of not being yelled at by every contributor.

- [x] Issue could be reproduced at least once
- [x] Issue could be reproduced by different players
- [x] Issue could be reproduced in multiple rounds
- [x] Issue happened in a recent (less than 7 days ago) round
- [x] [Couldn't find an existing issue about this](https://github.com/d3athrow/vgstation13/issues)

**Comment by SonixApache** [comment]
Heard complaints about ded serb
mommi said it was roid after 3 consecutive roidroids
tried to connect
dreamseeker crashed
it's been reported since about last saturday

**Comment by kilozombie** [comment]
game.ss13.moe breaks in general.

**Comment by ProfanedBane** [comment]
I can still access the logs fine at game.ss13.moe.

**Comment by kilozombie** [comment]
When the server becomes inaccessible, often so does game.ss13.moe/logs

**Comment by ProfanedBane** [comment]
For the purpose of log diving there was an instance of it on 14/08/18 at 15:19:12

**Comment by ProfanedBane** [comment]
This has just happened again, however people already playing are fine, but new connections are dropped or result in a dreamseeker crash.

**Comment by Exxion** [comment]
Yeah, every time this has happened, everything on game.ss13.moe goes down. It's a separate server from the rest of ss13.moe. (Except stats.ss13.moe, which is the same server as game.ss13.moe and also goes down.)
When it happens, most people are affected, but not all.
The server still responds to pings even to people who cannot connect.
I *think* the first time it happened, the server was on Deff, not Roid. I'm inclined to believe it has nothing to do with map, but I've been wrong before.

**Comment by Exxion** [comment]
#19246 had secret repo conflicts, which for some reason MoMMI failed to report
This was probably the cause
This has been fixed
Reopen if it happens again after next round

**Comment by kilozombie** [comment]
@Exxion That probably isn't the only PR with hidden secret repo conflicts
https://github.com/vgstation-coders/vgstation13/pull/19321 and https://github.com/vgstation-coders/vgstation13/pull/19320 are both completely nonfunctional and not in the game.

**Comment by Exxion** [comment]
Those are both after that one
The server just sticks with the last compiled version whenever it fails to compile
And #19246 failed to compile

**Comment by kilozombie** [comment]
Are you sure? Those PRs all fail to show up on the changelogs, but other changes do show up on the changelogs (and I'm fairly sure they're actually in the game)

**Comment by Exxion** [comment]
None from after #19246. In fact, none from several hours before it either, for some reason.
However I've just realized I am infinitely stupid, and the crashing started happening several days before that was merged
So reopening

EDIT: Actually looks like it was only a few minutes rather than hours before it that commits stopped being applied
So presumably all other commits from earlier in the same round

**Comment by kilozombie** [comment]
Currently happening again

**Comment by Pathid** [comment]
game.ss13.moe/logs and game.ss13.moe/minimaps/images/maps go down at the same time but not other parts of the site from what I can see.

**Comment by Exxion** [comment]
@Pathid You tried stats.ss13.moe and other parts of game.ss13.moe?

**Comment by Pathid** [comment]
stats.ss13.moe has been down since last month. ~~Other parts of the site seemed to work though.~~
Actually no, I didn't try any other parts of game.ss13.moe, just ss13.moe.

**Comment by Exxion** [comment]
In that case, see my first comment

**Comment by Pathid** [comment]
Yep. Skimmed over that.

**Comment by DamianX** [comment]
Happened again just now. Logs went offline too. Map was deff, so it's not exclusive to roidstation.

**Comment by SonixApache** [comment]
Probably not related, but my dreamseeker crashed 3 times in a row before actually working while trying to connect to my localhost for testing stuff.

**Comment by Inorien** [comment]
Doubt thats connected Sonix, if it worked eventually then its a different issue to this
So so far we have confirmed it only happening on roid and deff

**Comment by Faptastic** [comment]
Quite often a roid build freezes dreamdaemon for a longass time, you have the natural ui freeze from the single threading but this is easily double, triple the length of time needed before it returns to life.
Yet other times the very same compile will just work as normal. Is there any data on the state of DD during the connection issues? Can't connect to what isn't there afterall.

**Comment by SonixApache** [comment]
doing it again, on roid

**Comment by DamianX** [comment]
Right after I merged a PR. Could be dying because it's compiling?

**Comment by SonixApache** [comment]
Disappeared at some point in the round, so maybe.

**Comment by Kurfursten** [comment]
Happening RIGHT NOW on Box Station

Clearing Byond cached fixed it for me
Gear Icon->Preferences->Games->Clear Cache

**Comment by Exxion** [comment]
That's strange. Did you try accessing the logs, rapsheet, etc., @Kurfursten?

**Comment by Kurfursten** [comment]
This was the runtime log that got posted by Bobda
`[15:19:44] Skipped 45 runtimes in ,.`

I didn't try accessing any of those because I had heard of the cache fix and wanted to try it right away

**Comment by ProfanedBane** [comment]
Just had it happen mid round, I did some tests.

I got disconnected midway through a round with 40~ players on due to my ISP being a shit.
When trying to reconnect to the main server dreamseeker tries for about 2 minutes then crashes.
I can connect to over servers such as /tg/ and the test server, so it's definitely not a byond wide issue.
Game.ss13.moe and ss13.moe work fine, and if the test server is the same box as the main one it's likely not on pomfs end.

So what's causing dreamseeker to shit the bed?

**Comment by Exxion** [comment]
Test server is not the same box as the main server, but it's unusual that you could still connect to game.ss13.moe while being unable to connect to the main server

**Comment by SonixApache** [comment]
24-08-2018, ~2:30 AM UTC+2
happening again, map swapped from roid to roid a few minutes ago. Went from 33 players to 2 players. Clearing cache is not working.
I can't reach game.ss13.moe, and people were complaining that the minimaps were down before that.

**Comment by kilozombie** [comment]
Starting a new round seems to consistently fix it, even if it's from roid to roid
Before this gets fixed we need an intermediate way to make the server restart so it isn't unplayable for hours on end

**Comment by KingofHONK** [comment]
Forwarding a post from some random anon in the thread related to this issue:

Crazy theory about the crashing
I remember we used to have something like this and it turned out to be caused by a certain vault spawning
Did we get any new vaults roughly around the same time as roidstation?
Or could one certain vault interact badly with roidstation?
Just guessing at this point

Don't know shit about coder goobily gock but felt like it's worth forwarding over here

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Just happened when switching from bagel to deff (unofficially known as "shitmap bingo")

**Comment by kilozombie** [comment]
Thread also theorized that it is specifically this PR that causes it: https://github.com/vgstation-coders/vgstation13/pull/19139

**Comment by kilozombie** [comment]
I don't know of any reports happening before that PR and it seems to have started almost right on the money
Either way it's worth looking into exactly what point this stuff started

**Comment by Exxion** [comment]
That PR was several days before it started happening, but it's not impossible
Again, as has been pointed out multiple times in this issue, it's not specific to Roid, so that's not the tree I'd bark up

If I'm not mistaken, the first occurrence was on the tenth. Don't know what time, though.

**Comment by kilozombie** [comment]
Pomf seems to have fixed it. If so, we were all wrong, and it was him being essentially DDOSed by scraping bots, and adding a robots.txt to game.ss13.moe prevents it.

**Comment by SonixApache** [comment]
No way. Really?

**Comment by kilozombie** [comment]
![image](https://user-images.githubusercontent.com/6062247/44623955-8e8c0500-a893-11e8-873d-175125de50e6.png)

**Comment by Exxion** [comment]
I was aware of those posts, but I thought it happened again after that

**Comment by kilozombie** [comment]
Not sure, haven't heard about it.

**Comment by Exxion** [comment]
It's quite possible I was mistaken
I thought people said it had, but they were probably talking about something else
Let's hope that was the case

**Comment by kilozombie** [comment]
It's happening again @Exxion @SonixApache

**Comment by Exxion** [comment]
great

**Comment by Chazach** [comment]
UNO JUSTO
